core/rfb.js

```diff
@@ -1060,15 +1060,15 @@ export default class RFB extends EventTargetMixin {
         let pos = clientToElement(ev.clientX, ev.clientY,
                                   this._canvas);
 
+        let bmask = RFB.convertButtonMask(ev.buttons);
+
         switch (ev.type) {
             case 'mousedown':
                 setCapture(this._canvas);
-                this._handleMouseButton(pos.x, pos.y,
-                                        true, 1 << ev.button);
+                this._handleMouseButton(pos.x, pos.y, true, bmask);
```

Comment:
It's a bit weird keeping the `down` argument now that we have a full mask. But that's a cleanup I suppose we can do later.